libs/blocks/merch/merch.js

```diff
@@ -128,13 +128,42 @@ const GeoMap = {
   th_th: 'TH_th',
 };
 
+const LANG_STORE_PREFIX = 'langstore/';
+
+// eslint-disable-next-line consistent-return
+function getDefaultLangstoreCountry(language) {
+  if (language === 'en') {
+    return 'US';
+  }
+  if (language === 'ar') { // Arabic, not for Argentina
+    return 'DZ';
+  }
+  if (language === 'uk') { // Ukraine, not United Kingdom
+    return 'UA';
+  }
+  if (GeoMap[language]) {
+    return GeoMap[language].split('_', 2)[0];
+  }
+  for (const [, value] of Object.entries(GeoMap)) {
+    if (value.toLowerCase().endsWith(`_${language}`)) {
+      return value.split('_', 2)[0];
+    }
+  }
```

Comment:
so in the end you depend on the order of JSON object entries (which is not guaranteed) to return the correct language, correct?
In that case, i think i prefer a default language to locale map, setting en->us_EN, fr->fr_FR, etc...
then you can do something like `GeoMap[language] || LangDefaultMap[language.split('_', 2)[0]] `

Reply:
@npeltier  I created `LanguageMap` for mappings where language code is not equal to the country code : 
`en -> US, el -> GR, ja -> JP` etc.
For others I simply us language code as country code if it exists in `GeoMap`
There are also some strange language variants in the langstore like es-419, pt-PT that I need to split and then do the same.